mpags-cipher.cpp

```diff
@@ -0,0 +1,47 @@
+#include <iostream>
+#include <string>
+#include <vector>
+
+int main(int argc, char* argv[])
+{
+    const std::vector<std::string> cmdLineArgs { argv, argv+argc };
+    const size_t n = cmdLineArgs.size();
+
+    for (std::size_t i{0}; i < n; i++) {
+        // print help message and version information
+        if (cmdLineArgs[i] == "-h" || cmdLineArgs[i] == "--help") {
+            std::cout << "some help messages" << std::endl;
+        }
+
+        else if (cmdLineArgs[i] == "--version") {
+            std::cout << "version 0.1.0" << std::endl;
+        }
```

Comment:
Conventionally, the program would exit after printing either the help or the version.